src/types/sigstore/validate.ts

```diff
@@ -0,0 +1,88 @@
+import { InvalidBundleError } from '../../error';
+import { WithRequired } from '../utility';
+import { Bundle, VerificationMaterial } from './__generated__/sigstore_bundle';
+import { MessageSignature } from './__generated__/sigstore_common';
+
+// Sigstore bundle with all required fields populated
+type ValidBundle = Bundle & {
+  verificationMaterial: VerificationMaterial & {
+    content: NonNullable<VerificationMaterial['content']>;
+  };
+  content:
+    | (Extract<Bundle['content'], { $case: 'messageSignature' }> & {
+        messageSignature: WithRequired<MessageSignature, 'messageDigest'>;
+      })
+    | Extract<Bundle['content'], { $case: 'dsseEnvelope' }>;
+};
+
+// Performs basic validation of a Sigstore bundle to ensure that all required
+// fields are populated. This is not a complete validation of the bundle, but
+// rather a check that the bundle is in a valid state to be processed by the
+// rest of the code.
+export function assertValidBundle(b: Bundle): asserts b is ValidBundle {
+  // Content-related validation
+  if (b.content === undefined) {
+    throw new InvalidBundleError('content is undefined');
+  }
+
+  switch (b.content.$case) {
+    case 'messageSignature':
+      if (b.content.messageSignature.messageDigest === undefined) {
+        throw new InvalidBundleError('messageDigest is undefined');
```

Comment:
Is there a way you could use full property paths (e.g. messageSignature.messageDigest) in the error message?
Is there a way you could mention all the missing fields rather than stopping on the first missing one?

Reply:
Good suggestions. Done!